spec/jobs/reports/idv_legacy_conversion_supplement_report_spec.rb

```diff
@@ -0,0 +1,333 @@
+          )
+        end
+
+        let(:iaa1) { 'iaa1' }
+
+        let(:iaa1_sp) do
+          create(
+            :service_provider,
+            iaa: iaa1,
+            iaa_start_date: iaa1_range.begin,
+            iaa_end_date: iaa1_range.end,
+          )
+        end
+
+        let(:iaa_order1) do
+          build_iaa_order(order_number: 1, date_range: iaa1_range, iaa_gtc: gtc1)
+        end
+
+        let(:inside_iaa1) { iaa1_range.begin + 1.day }
+
+        let(:user1) { create(:user) }
+
+        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
+
+        before do
+          iaa_order1.integrations << build_integration(
+            issuer: iaa1_sp.issuer,
+            partner_account: partner_account1,
+          )
+          iaa_order1.save
+          create(
+            :sp_upgraded_biometric_profile,
+            issuer: iaa1_sp.issuer, user_id: user1.id, upgraded_at: inside_iaa1,
+          )
+        end
+
+        it 'finds the iaa related to the upgraded profile' do
+          expect(csv.length).to eq(1)
+          aggregate_failures do
+            row = csv.find { |r| r['issuer'] == iaa1_sp.issuer }
+
+            expect(row.to_h.symbolize_keys).to eq(
+              {
+                iaa_order_number: 'gtc1234-0001',
+                iaa_start_date: '2020-04-15',
+                iaa_end_date: '2021-04-14',
+                issuer: iaa1_sp.issuer,
+                friendly_name: iaa1_sp.friendly_name,
+                year_month: '202004',
+                year_month_readable: 'April 2020',
+                user_count: '1',
+              },
+            )
+          end
+        end
+      end
+
+      context 'when there are converted profiles with two issuers in September 2020' do
+        let(:partner_account2) { create(:partner_account) }
+        let(:iaa2_range) { DateTime.new(2020, 9, 1).utc..DateTime.new(2021, 8, 30).utc }
+
+        let(:gtc2) do
+          create(
+            :iaa_gtc,
+            gtc_number: 'gtc5678',
+            partner_account: partner_account2,
+            start_date: iaa2_range.begin,
+            end_date: iaa2_range.end,
+          )
+        end
+
+        let(:iaa2) { 'iaa2' }
+
+        let(:iaa2_sp1) do
+          create(
+            :service_provider,
+            iaa: iaa2,
+            iaa_start_date: iaa2_range.begin,
+            iaa_end_date: iaa2_range.end,
+          )
+        end
+
+        let(:iaa2_sp2) do
+          create(
+            :service_provider,
+            iaa: iaa2,
+            iaa_start_date: iaa2_range.begin,
+            iaa_end_date: iaa2_range.end,
+          )
+        end
+
+        let(:iaa_order2) do
+          build_iaa_order(order_number: 2, date_range: iaa2_range, iaa_gtc: gtc2)
+        end
+
+        let(:inside_iaa2) { iaa2_range.begin + 1.day }
+
+        let(:user1) { create(:user) }
+
+        let(:user2) { create(:user) }
+
+        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
```

Comment:
could probably remove some of these extra liens
```suggestion
        let(:inside_iaa2) { iaa2_range.begin + 1.day }
        let(:user1) { create(:user) }
        let(:user2) { create(:user) }
        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
```